### PR TITLE
Added Test-cases InputValidator Plugin

### PR DESCRIPTION
**Pull request by tasbi03**
## Description

This pull request introduces comprehensive unit tests and configurations for the **`input-validator`** project, improving the test coverage, validating core functionality, and enhancing code quality and reliability.

---

### Added Unit Tests:
- **Core Functionality**:
  - **`validateFile.test.ts`**: Tests various cases for file validation (valid, invalid, empty, and oversized files).
  - **`getValidator.test.ts`**: Ensures the `getValidator` utility retrieves correct validators or returns `undefined` for invalid cases.
  - **`getPromptStyles.test.ts`**: Validates style generation for different prompt types (`error`, `success`, etc.).
  - **`mergePluginConfig.test.ts`**: Ensures user configurations are merged properly with default plugin configurations.
  - **`useRcbPlugin.test.ts`**: Tests plugin behavior including text input, file upload validations, and event handling.

- **Configuration Files**:
  - **`jest.config.js`**: Configures Jest for testing.
  - **`setup.jest.js`**: Adds polyfills (e.g., `structuredClone`) and sets up the environment for consistent test execution.

---

Closes issue #3 
---
## What is the proposed approach?

The PR covers:
1. **Comprehensive Testing**:
   - Tests edge cases for file and text validation logic.
   - Ensures utility functions (`mergePluginConfig`, `getValidator`, `getPromptStyles`) behave correctly under various configurations.
   - Validates plugin behavior including event listeners and style restoration.
2. **Improved Code Coverage**:
   - Focused on untested lines and branches to achieve near-complete coverage.
   - Adds tests for scenarios like `null`/`undefined` inputs, event listener behavior, and default configurations.
3. **Environment Setup**:
   - Configures Jest and introduces `setup.jest.js` to provide consistent polyfills for `structuredClone` and other necessary utilities.

---

## Checklist:

- [x] Added unit tests for `validateFile`, `getValidator`, `mergePluginConfig`, `getPromptStyles`, and `useRcbPlugin`.
- [x] Configured `jest.config.js` for comprehensive test runs.
- [x] Introduced `setup.jest.js` for environment consistency.
- [x] Verified test coverage and addressed untested lines or branches.

---

## Testing Summary:

This PR ensures:
- Core functionality and utility methods are validated against edge cases.
- Styles and plugin configurations are correctly managed.
- Event listeners in `useRcbPlugin` behave as expected for text inputs and file uploads.

---

Feel free to review and suggest additional improvements or scenarios for coverage.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,14 +48,22 @@
   },
   "devDependencies": {
     "@eslint/js": "^9.11.1",
+    "@testing-library/jest-dom": "^6.6.3",
+    "@testing-library/react": "^16.0.1",
+    "@types/jest": "^29.5.14",
     "@types/react": "^18.3.10",
     "@types/react-dom": "^18.3.0",
+    "@types/testing-library__jest-dom": "^5.14.9",
```
Comment (tjtanjin) [inline package.json]: We do not need this, refer here: https://www.npmjs.com/package/@types/testing-library__jest-dom

**Review by tjtanjin — CHANGES_REQUESTED** [review]
Hey @tasbi03! Thanks for the PR, the test cases look great, though a couple of stuffs to cleanup:
- Lint errors need to be fixed
- Remove unnecessary packages in package file
- Add the coverage folder to gitignore (coverage files shouldn't be committed)

**Review by tjtanjin — APPROVED** [review]
